pkg/apiserver/api_installer.go

```diff
@@ -689,6 +709,9 @@ var _ ScopeNamer = legacyScopeNaming{}
 
 // Namespace returns the namespace from the query or the default.
 func (n legacyScopeNaming) Namespace(req *restful.Request) (namespace string, err error) {
+	if n.scope.Name() == meta.RESTScopeNameRoot {
```

Comment:
should this block only be done in the `if !ok || len(values) == 0 {` case? Otherwise a request to `foo?namespace=bar` is treated as namespace none